crates/torii/src/cli.rs

```diff
@@ -78,13 +165,35 @@ async fn main() -> anyhow::Result<()> {
 
     let provider = JsonRpcClient::new(HttpTransport::new(Url::parse(&args.rpc).unwrap()));
 
-    let manifest = if let Some(manifest_path) = args.manifest {
-        Manifest::load_from_path(manifest_path).expect("Failed to load manifest")
+    let mut manifest_path = scarb::ops::find_manifest_path(args.manifest.as_deref())?;
+    let config = Config::builder(manifest_path.clone())
+        .log_filter_directive(env::var_os("SCARB_LOG"))
+        .build()?;
+    let env_metadata = if config.manifest_path().exists() {
+        let ws = scarb::ops::read_workspace(config.manifest_path(), &config)?;
+
+        // TODO: Check the updated scarb way to read profile specific values
+        dojo_metadata_from_workspace(&ws).and_then(|inner| inner.env().cloned())
     } else {
-        Manifest::default()
+        None
+    };
+
+    println!("{:?}", env_metadata);
+
+    if manifest_path.ends_with("Scarb.toml") {
+        manifest_path.pop();
+        manifest_path.push("target/dev/manifest.json");
```

Comment:
While we're already have the `scarb`'s `Workspace`, we can avoid hardcoding and generalize the path to `manifest.json` by using `Workspace::target_dir()`. Can refer to how it's done for `sozo migrate` here: 

(although we should propagate the error instead of unwrapping)

https://github.com/dojoengine/dojo/blob/2caa5537a248619526cd2f8e0c57025736aade74/crates/sozo/src/commands/migrate.rs#L37-L40